### PR TITLE
Stabilize ConsoleApp sample

### PR DESCRIPTION
**Pull request by Kielek**
It allows to send traces in the very base scenario, described in DEVELOPING.md

Changes can be reverted when #321 will be solved and we will be able to send traces even when the unhandled exception occurs.

Worth to include to make the potential new-developer easier start. I have spent to much time to detect what is wrong.

### CODE REVIEW

**Comment by nrcventura** [comment]
Can you explain a little more how these changes make the sample app more stable? If you are referring to telemetry loss, it's unclear to me how these changes help that scenario. If there's an unhandled exception, wouldn't a try catch block in the sample app catch that?

**Comment by Kielek** [comment]
@nrcventura, main idea was to avoid unhandled exception by using _working_ endpoints.
After changes it is working even without access to the internet.

Sample log from collector:
```log
dev-otel-collector-1  | 2022-01-21T06:12:07.428Z        INFO    loggingexporter/logging_exporter.go:40  TracesExporter  {"#spans": 4}
dev-otel-collector-1  | 2022-01-21T06:12:07.428Z        DEBUG   loggingexporter/logging_exporter.go:49  ResourceSpans #0
dev-otel-collector-1  | Resource SchemaURL:
dev-otel-collector-1  | Resource labels:
dev-otel-collector-1  |      -> service.name: STRING(unknown_service:ConsoleApp)
dev-otel-collector-1  | InstrumentationLibrarySpans #0
dev-otel-collector-1  | InstrumentationLibraryMetrics SchemaURL:
dev-otel-collector-1  | InstrumentationLibrary OpenTelemetry.ClrProfiler.ConsoleApp
dev-otel-collector-1  | Span #0
dev-otel-collector-1  |     Trace ID       : 032dbaf43cf597a69101001147cf48a9
dev-otel-collector-1  |     Parent ID      :
dev-otel-collector-1  |     ID             : 3c2f3c84f14645a0
dev-otel-collector-1  |     Name           : Main
dev-otel-collector-1  |     Kind           : SPAN_KIND_UNSPECIFIED
dev-otel-collector-1  |     Start time     : 2022-01-21 06:12:01.481235 +0000 UTC
dev-otel-collector-1  |     End time       : 2022-01-21 06:12:07.268744 +0000 UTC
dev-otel-collector-1  |     Status code    : STATUS_CODE_UNSET
dev-otel-collector-1  |     Status message :
dev-otel-collector-1  | Attributes:
dev-otel-collector-1  |      -> net.host.ip: STRING(192.168.0.113)
dev-otel-collector-1  | Span #1
dev-otel-collector-1  |     Trace ID       : 032dbaf43cf597a69101001147cf48a9
dev-otel-collector-1  |     Parent ID      : 8e774cf5781f3db5
dev-otel-collector-1  |     ID             : 7c8766a87c11a863
dev-otel-collector-1  |     Name           : RunAsync
dev-otel-collector-1  |     Kind           : SPAN_KIND_UNSPECIFIED
dev-otel-collector-1  |     Start time     : 2022-01-21 06:12:01.496828 +0000 UTC
dev-otel-collector-1  |     End time       : 2022-01-21 06:12:06.567778 +0000 UTC
dev-otel-collector-1  |     Status code    : STATUS_CODE_UNSET
dev-otel-collector-1  |     Status message :
dev-otel-collector-1  | Attributes:
dev-otel-collector-1  |      -> foo: STRING(bar)
dev-otel-collector-1  |      -> net.host.ip: STRING(192.168.0.113)
dev-otel-collector-1  | InstrumentationLibrarySpans #1
dev-otel-collector-1  | InstrumentationLibraryMetrics SchemaURL:
dev-otel-collector-1  | InstrumentationLibrary OpenTelemetry.ClrProfiler.OpenTracingShim
dev-otel-collector-1  | Span #0
dev-otel-collector-1  |     Trace ID       : 032dbaf43cf597a69101001147cf48a9
dev-otel-collector-1  |     Parent ID      : 3c2f3c84f14645a0
dev-otel-collector-1  |     ID             : 8e774cf5781f3db5
dev-otel-collector-1  |     Name           : OpenTracing Span
dev-otel-collector-1  |     Kind           : SPAN_KIND_CLIENT
dev-otel-collector-1  |     Start time     : 2022-01-21 06:12:01.491754 +0000 UTC
dev-otel-collector-1  |     End time       : 2022-01-21 06:12:07.266831 +0000 UTC
dev-otel-collector-1  |     Status code    : STATUS_CODE_UNSET
dev-otel-collector-1  |     Status message :
dev-otel-collector-1  | Attributes:
dev-otel-collector-1  |      -> action.success: STRING(true)
dev-otel-collector-1  |      -> net.host.ip: STRING(192.168.0.113)
dev-otel-collector-1  | Events:
dev-otel-collector-1  | SpanEvent #0
dev-otel-collector-1  |      -> Name: action success
dev-otel-collector-1  |      -> Timestamp: 2022-01-21 06:12:07.26621 +0000 UTC
dev-otel-collector-1  |      -> DroppedAttributesCount: 0
dev-otel-collector-1  | Span #1
dev-otel-collector-1  |     Trace ID       : 032dbaf43cf597a69101001147cf48a9
dev-otel-collector-1  |     Parent ID      : 8e774cf5781f3db5
dev-otel-collector-1  |     ID             : 26f42c0679dabafa
dev-otel-collector-1  |     Name           : Client POST https://www.example.com/
dev-otel-collector-1  |     Kind           : SPAN_KIND_CLIENT
dev-otel-collector-1  |     Start time     : 2022-01-21 06:12:06.584377 +0000 UTC
dev-otel-collector-1  |     End time       : 2022-01-21 06:12:07.265354 +0000 UTC
dev-otel-collector-1  |     Status code    : STATUS_CODE_UNSET
dev-otel-collector-1  |     Status message :
dev-otel-collector-1  | Attributes:
dev-otel-collector-1  |      -> response.length: STRING(1256)
dev-otel-collector-1  |      -> http.status_code: STRING(200)
dev-otel-collector-1  |      -> net.host.ip: STRING(192.168.0.113)
dev-otel-collector-1  |
```

**Review by pellared — COMMENTED** [review]
I would love to have it merged soon. The problem that this PR solves would help me also in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/324

I put only 2 suggestions that I would love to get addressed as I think this sample gets overcomplicated. 

In the future, we should probably split this sample application and add `README.md` (or comments in code) describing what these samples are demonstrating.

File: samples/ConsoleApp/Program.cs [hunk 1]
```diff
@@ -58,21 +60,56 @@ private static async Task RunAsync()
...
-                scope.Span.SetTag("response.length", responseContent.Length);
+                if (statusCode.HasValue)
+                {
+                    scope.Span.SetTag("http.status_code", statusCode.Value);
+                }
+
+                if (responseLength.HasValue)
+                {
+                    scope.Span.SetTag("response.length", responseLength.Value);
+                }
```
[the remark this answers is not in the document]
Comment (Kielek) [reply samples/ConsoleApp/Program.cs]: Sure, the goal was to fix this app, but I simplified it a bit.

File: samples/ConsoleApp/Program.cs [hunk 2]
```diff
@@ -58,21 +60,56 @@ private static async Task RunAsync()
...
             }
         }
 
+        private static void LogHttpRequestException(string requestUrl, HttpRequestException httpRequestException)
+        {
+            Console.WriteLine($"HttpRequestException occurred while calling {requestUrl}, {httpRequestException}");
+        }
```
[the remark this answers is not in the document]
Comment (Kielek) [reply samples/ConsoleApp/Program.cs]: Personally, I do not think that extraction process could be harder with this method. Fixed.

File: samples/ConsoleApp/Program.cs [hunk 3]
```diff
@@ -58,21 +59,34 @@ private static async Task RunAsync()
                     request.Headers.Add(kvp.Key, kvp.Value);
                 }
 
-                using var response = await client.SendAsync(request);
+                try
+                {
+                    using var response = await client.SendAsync(request);
 
-                scope.Span.SetTag("http.status_code", (int)response.StatusCode);
+                    _ = await response.Content.ReadAsStringAsync();
```
Comment (pellared) [inline samples/ConsoleApp/Program.cs:66]:
```suggestion
                    await response.Content.ReadAsStringAsync();
```

**Review by pellared — APPROVED** [review]

**Review by nrcventura — APPROVED** [review]

**Pull request merged**